### DISCUSSION TITLE
Функция Source.getArtistsAlbums возвращает недоступные альбомы

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#69] [Помощь с ошибкой] [answered]
Использую у себя шаблон "[Назад в этот день](https://github.com/Chimildic/goofy/discussions/12#discussion-78656)"
Сегодня заметил, что среди альбомов повился недоступный для прослушивания альбом.
5g3oqy7nAK03ke2cR5c95I - Metallica Live In Argentina (1993 – 2017)

Если, я правильно понял, API [Get an Artist's Albums](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-an-artists-albums) возвращает все альбомы, если они доступны хотя бы в одном регионе. 
И на результат никак не влияет указание параметра `market`. Даже если указывать `market = from_token`, всё равно в результатах могут оказаться недоступные в моем регионе альбомы. И будет указано `"is_playable": true`

Не придумал ничего лучше, чем подправить функцию `getArtistsAlbums`. Убрал `market` из запроса.
```js
let path = `artists/${artist.id}/albums?include_groups=${groups}&limit=50`;
```
Иначе в результатах нет `"available_markets"`.

Плюс, добавил себе еще одну функцию `removeUnavailable` для фильтрации именно альбомов.

Или я что не так понял и есть более простой выход?

**Comment by Chimildic** [comment, marked as answer]
Больше похоже на баг Spotify. Альбом `is_playable: true`, но ни один трек в нем недоступен. 
Но это не должно мешать последующему коду. Тест показывает, что стандартная `removeUnavailable` не пропустит эти треки дальше.
```js
let tracks = Source.getAlbumTracks({id: '5g3oqy7nAK03ke2cR5c95I'})
Filter.removeUnavailable(tracks);
console.log(tracks.length) // будет ноль
```

**Reply by efremych** [reply]
Да, тоже кажется, что похоже на баг. Иначе, какая-то странная реализация. Хотел поделиться информацией.
Фильтровать треки можно, да, просто пытаюсь экономить запросы :)

**Reply by Chimildic** [reply]
Мне кажется, не то место для экономии. На 50 треков `removeUnavailable` сделает 1 запрос в данном случаи. В сравнении с некоторым функциями, это очень мало.
